packages/contracts/contracts/governance/MintManager.sol

```diff
@@ -0,0 +1,230 @@
+     */
+    address[] internal recipients;
+
+    /**
+     * @notice A mapping of the recipient's share.
+     */
+    mapping(address => uint256) public shareOf;
+
+    /**
+     * @notice The number of the last block that minted tokens.
+     */
+    uint256 internal lastMintedBlock;
+
+    /**
+     * @notice Reverts if the caller is not a L1Block contract.
+     */
+    modifier onlyL1Block() {
+        require(msg.sender == Predeploys.L1_BLOCK_ATTRIBUTES, "MintManager: only the L1Block can call this function");
+        _;
+    }
+
+    /**
+     * @notice Semantic version.
+     * @custom:semver 1.0.0
+     */
+    string public constant version = "1.0.0";
+
+    /**
+     * @notice Constructs the MintManager contract.
+     *
+     * @param _mintActivatedBlock  The number of L2 block number which the mint function is activated.
+     * @param _initMintPerBlock    The amount of the initial minting per block.
+     * @param _slidingWindowBlocks The number of blocks per epoch.
+     * @param _decayingFactor      The decaying factor that reduces the amount of minting.
+     */
+    constructor(
+        uint256 _mintActivatedBlock,
+        uint256 _initMintPerBlock,
+        uint256 _slidingWindowBlocks,
+        uint256 _decayingFactor
+    ) {
+        MINT_ACTIVATED_BLOCK = _mintActivatedBlock;
+        INIT_MINT_PER_BLOCK = _initMintPerBlock;
+        SLIDING_WINDOW_BLOCKS = _slidingWindowBlocks;
+        DECAYING_FACTOR = _decayingFactor;
+
+        GOVERNANCE_TOKEN = GovernanceToken(Predeploys.GOVERNANCE_TOKEN);
+    }
+
+    /**
+     * @notice Initializer to settings recipients.
+     *
+     * @param _recipients List of the recipient.
+     * @param _shares     List of token distribution ratios for each recipient.
+     */
+    function initialize(
+        address[] calldata _recipients,
+        uint256[] calldata _shares
+    ) external initializer {
+        require(_recipients.length == _shares.length, "MintManager: invalid length of array");
+
+        uint256 totalShares = 0;
+        for (uint256 i = 0; i < _recipients.length; i++) {
+            address recipient = _recipients[i];
+            require(recipient != address(0), "MintManager: recipient address cannot be 0");
+            uint256 share = _shares[i];
+            require(share != 0, "MintManager: share cannot be 0");
+            totalShares += share;
+
+            recipients.push(recipient);
+            shareOf[recipient] = share;
+        }
+        require(totalShares == SHARE_DENOMINATOR, "MintManager: invalid shares");
+    }
+
+    /**
+     * @notice Mints and distributes tokens.
+     *         It must be called every block and cannot be called more than once in the same block.
+     *         If it is the first time tokens are being minted, mint the amount of tokens that
+     *         should be minted up to the current block number.
+     */
+    function mint() external onlyL1Block {
+        if (MINT_ACTIVATED_BLOCK > block.number) {
```

Comment:
Did you intend not more or equal, strictly more than?
```suggestion
        if (MINT_ACTIVATED_BLOCK >= block.number) {
```

Reply:
`MINT_ACTIVATED_BLOCK > block.number` is right. 
The block number `MINT_ACTIVATED_BLOCK ` should also be minted.